core/src/sumeragi/mod.rs

```diff
@@ -56,15 +57,20 @@ trait Consensus {
 }
 
 /// `Sumeragi` is the implementation of the consensus.
+#[allow(clippy::mutex_integer)]
 #[derive(Debug)]
 pub struct Sumeragi {
     internal: SumeragiWithFault<NoFault>,
     config: Configuration,
+    metrics_mutex: Mutex<Metrics>,
+    last_block_height: Mutex<u64>,
+    last_metric_tx_amounts: Mutex<f64>,
+    last_metric_tx_amounts_counter: Mutex<u64>,
```

Comment:
Why not atomic?

Reply:
To me, this type of problem is not what "atomics are for". If you see that it's a mutex you directly understand the type of interaction that is going on. Maybe these three should be consolidated into one struct that sits inside the mutex instead.

Reply:
Ok, lets wait for other reviewers opinions on this one
Initially [clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#arithmetic) brought my attention to this one.

Reply:
> To me, this type of problem is not what "atomics are for"
I'm afraid this is precisely what atomics are for, synchronised interior mutability of shared data. 

All you do by wrapping these objects into individual mutexes is confuse the reader. 

- If you had a structure behind a mutex like you suggested, the reader knows that having partially updated the variables inside the structure is inconsistent state. `RWLock` is the exact same thing, but more complex and useful if you read frequently, and write infrequently. 

- If you have three atomics, it means just that: you need to mutate the numbers, and you share the data between different threads. Data is completely unrelated. 

- If you have a RefCell, that means that data races are impossible. All you want is to make the borrow checker happy.

Reply:
So please @SamHSmith consider using atomics and using comments to convey the intent, instead of $6 \times$ the amount of space that's actually needed.

Reply:
"A mutex muddies the intent. If you need just the interior mutability in a single-threaded context, use RefCell. If you need just the thread-safe separate modifications of the objects use a atomics. If you need them to be modified at the same time, and the values need to be in sync with each other, you need a structure."

I would prefer to have a struct inside a RefCell. But the struct lives inside an Arc so rust says no.

Reply:
> I would prefer to have a struct inside a RefCell. But the struct lives inside an Arc so rust says no.

It should be possible to remove the `Arc` and instead spawn a single async task to scan the `isi` and update the metrics, so not even a `RefCell` would be needed.